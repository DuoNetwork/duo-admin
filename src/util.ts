import moment from 'moment';
import request from 'request';
import * as CST from './constants';
import { IOption } from './types';

class Util {
	public log(text: any): void {
		console.log(moment().format('HH:mm:ss.SSS') + ' ' + text);
	}

	// public truncateNum(num: number, decimal: number = 3): number {
	// 	let x = num.toString();
	// 	x = x.slice(0, x.indexOf('.') + decimal);
	// 	return Number(x);
	// }

	public get(url: string, headerOthers?: object): Promise<any> {
		return new Promise((resolve, reject) =>
			request(
				{
					url,
					headers: Object.assign(
						{
							'user-agent': 'node.js'
						},
						headerOthers || {}

					)
				},
				(error, res, body) => {
					if (error) reject(error);
					else if (res.statusCode === 200) resolve(body);
					else reject('Error status ' + res.statusCode + ' ' + res.statusMessage);
				}
			)
		);
	}

	public generateRandomIdx(max: number, alpha: number): number[] {
		const output: number[] = [];
		if (alpha === 1) for (let i = 0; i < max; i++) output.push(i);
		else {
			const num = Math.floor(max * alpha);

			while (output.length < num) {
				const item = Math.floor(Math.random() * max);
				if (output.indexOf(item) < 0) output.push(item);
			}
		}

		return output;
	}

	public postJson(url: string, json: object): Promise<object> {
		return new Promise((resolve, reject) =>
			request(
				{
					url,
					method: 'POST',
					headers: {
						'Content-Type': 'application/json'
					},
					json
				},
				(error, res, body) => {
					if (error) reject(error);
					else if (res.statusCode === 200) resolve(body);
					else reject('Error status ' + res.statusCode + ' ' + res.statusMessage);
				}
			)
		);
	}

	public parseOptions(argv: string[]): IOption {
		const option = {
			live: process.argv.includes('live'),
			dynamo: process.argv.includes('dynamo'),
			aws: process.argv.includes('aws'),
			gcp: process.argv.includes('gcp'),
			azure: process.argv.includes('azure'),
			force: process.argv.includes('force'),
			gasPrice: 5e9,
			gasLimit: 200000,
			eth: 0,
			address: '',
			privateKey: '',
			price: 0,
			source: '',
			pwd: '',
			event: '',
			provider: '',
			contractState: '',
			accountNum: 1,
			saveAccount: false,
			from: '',
			to: '',
			value: 0,
			index: 0,
			total: 10,
			minEther: 0.02,
			alpha: 0.3,
			amtA: 0,
			amtB: 0,
			generator: ''
		};
		for (let i = 3; i < argv.length; i++) {
			const args = argv[i].split('=');
			switch (args[0]) {
				case 'gasPrice':
					option.gasPrice = Number(args[1]) || option.gasPrice;
					break;
				case 'gasLimit':
					option.gasLimit = Number(args[1]) || option.gasLimit;
					break;
				case 'eth':
					option.eth = Number(args[1]) || option.eth;
					break;
				case 'address':
					option.address = args[1] || option.address;
					break;
				case 'privateKey':
					option.privateKey = args[1] || option.privateKey;
					break;
				case 'price':
					option.price = Number(args[1]) || option.price;
					break;
				case 'pwd':
					option.pwd = args[1] || option.pwd;
					break;
				case 'source':
					option.source = args[1] || option.source;
					break;
				case 'event':
					option.event = args[1] || option.event;
					break;
				case 'provider':
					option.provider = args[1] || option.provider;
					break;
				case 'contractState':
					option.contractState = args[1] || option.contractState;
					break;
				case 'accountNum':
					option.accountNum = Number(args[1]) || option.accountNum;
					break;
				case 'saveAccount':
					option.saveAccount = args[1] === 'yes' || option.saveAccount;
					break;
				case 'from':
					option.from = args[1] || option.from;
					break;
				case 'total':
					option.total = Number(args[1]) || option.total;
					break;
				case 'minEther':
					option.minEther = Number(args[1]) || option.minEther;
					break;
				case 'alpha':
					option.alpha = Number(args[1]) || option.alpha;
					break;
				case 'amtA':
					option.amtA = Number(args[1]) || option.amtA;
					break;
				case 'amtB':
					option.amtB = Number(args[1]) || option.amtB;
					break;
				case 'value':
					option.value = Number(args[1]) || option.value;
					break;
				case 'index':
					option.index = Number(args[1]) || option.index;
					break;
				case 'generator':
					option.generator = args[1] || option.generator;
					break;
				default:
					break;
			}
		}

		if (!option.provider)
			if (option.source === CST.SRC_MYETHER)
				option.provider = option.live
					? CST.PROVIDER_MYETHER_LIVE
					: CST.PROVIDER_MYETHER_DEV;
			else if (option.source === CST.SRC_INFURA)
				option.provider = option.live ? CST.PROVIDER_INFURA_LIVE : CST.PROVIDER_INFURA_DEV;
			else {
				option.provider = CST.PROVIDER_LOCAL_WS;
				option.source = '';
			}

		return option;
	}

	public getDynamoRole(tool: string, useDynamo: boolean): string {
		switch (tool) {
			case 'bitfinex':
			case 'gemini':
			case 'kraken':
			case 'gdax':
				return useDynamo ? CST.AWS_DYNAMO_ROLE_TRADE : CST.AWS_DYNAMO_ROLE_STATUS;
			case 'commit':
			case 'node':
				return CST.AWS_DYNAMO_ROLE_STATUS;
			case 'subscribe':
				return CST.AWS_DYNAMO_ROLE_EVENT;
			case 'hourly':
				return CST.AWS_DYNAMO_ROLE_HOURLY;
			case 'minutely':
				return CST.AWS_DYNAMO_ROLE_MINUTELY;
			default:
				return '';
		}
	}

	public getStatusProcess(tool: string, option: IOption) {
		let type = '';
		const platform = option.azure ? '_AZURE' : option.gcp ? '_GCP' : '_AWS';
		const privacy = option.dynamo ? '_PUBLIC' : '_PRIVATE';
		let source = '';

		switch (tool) {
			case 'bitfinex':
			case 'gemini':
			case 'kraken':
			case 'gdax':
				type = 'PRICE';
				source = '_' + tool.toUpperCase();
				break;
			case 'subscribe':
				type = 'EVENT';
				source =
					'_' +
					([CST.EVENT_START_PRE_RESET, CST.EVENT_START_RESET].includes(option.event)
						? option.event.toUpperCase()
						: 'OTHERS');
				break;
			case 'commit':
				type = 'FEED';
				break;
			case 'hourly':
				type = 'HOURLY';
				break;
			case 'minutely':
				type = 'MINUTELY';
				break;
			case 'node':
				type = 'CHAIN';
				break;
			default:
				return '';
		}

		return type + platform + privacy + source;
	}

	public getNowTimestamp() {
		return moment().valueOf();
	}

	public isNumber(input: any): boolean {
		const num = Number(input);
		return isFinite(num) && !isNaN(num);
	}

	public isEmptyObject(obj: object | undefined | null): boolean {
		if (!obj) return true;

		for (const prop in obj) if (obj.hasOwnProperty(prop)) return false;

		return true;
	}
}

const util = new Util();
export default util;
