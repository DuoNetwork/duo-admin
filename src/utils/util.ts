import moment from 'moment';
import * as CST from '../common/constants';
import { IOption } from '../common/types';
// import { Options } from 'aws-cli-js';

class Util {
	public logLevel: string = CST.LOG_INFO;

	private log(text: any, level?: string): void {
		if (level && CST.LOG_RANKING[this.logLevel] >= CST.LOG_RANKING[level])
			console.log(`${moment().format('HH:mm:ss.SSS')} [${level}]: ` + text);
	}

	public logInfo(text: any): void {
		this.log(text, CST.LOG_INFO);
	}

	public logDebug(text: any): void {
		this.log(text, CST.LOG_DEBUG);
	}

	public logError(text: any): void {
		this.log(text, CST.LOG_ERROR);
	}

	public composeQuery(paras: { [key: string]: any }): string {
		let query: string = '?';
		for (const key in paras) query += key + '=' + paras[key] + '&';
		query = query.slice(0, -1);
		return query;
	}

	public parseOptions(argv: string[]): IOption {
		const option = {
			forceREST: argv.includes('rest'),
			live: process.argv.includes('live'),
			dbLive: process.argv.includes('dbLive'),
			server: process.argv.includes('server'),
			dynamo: process.argv.includes('dynamo'),
			aws: process.argv.includes('aws'),
			gcp: process.argv.includes('gcp'),
			azure: process.argv.includes('azure'),
			force: process.argv.includes('force'),
			pair: '',
			contractType: 'Beethoven',
			tenor: 'Perpetual',
			assets: [''],
			sources: [''],
			exSources: [''],
			gasPrice: 5e9,
			gasLimit: 200000,
			source: '',
			event: '',
			provider: '',
			period: 1,
			base: 'USD',
			quote: 'ETH'
		};
		for (let i = 3; i < argv.length; i++) {
			const args = argv[i].split('=');
			switch (args[0]) {
				case 'exSources':
					option.exSources = args[1].split(',');
					break;
				case 'sources':
					option.sources = args[1].split(',');
					break;
				case 'pair':
					option.pair = args[1].replace('_', '|') || option.pair;
					break;
				case 'assets':
					option.assets = args[1].split(',');
					break;
				case 'gasPrice':
					option.gasPrice = Number(args[1]) || option.gasPrice;
					break;
				case 'gasLimit':
					option.gasLimit = Number(args[1]) || option.gasLimit;
					break;
				case 'source':
					option.source = args[1] || option.source;
					break;
				case 'event':
					option.event = args[1] || option.event;
					break;
				case 'provider':
					option.provider = args[1] || option.provider;
					break;
				case 'period':
					option.period = Number(args[1]);
					break;
				case 'base':
					option.base = args[1] || option.base;
					break;
				case 'quote':
					option.quote = args[1] || option.quote;
					break;
				case 'contractType':
					option.contractType = args[1] || option.contractType;
					break;
				case 'tenor':
					option.tenor = args[1] || option.tenor;
					break;
				default:
					break;
			}
		}

		return option;
	}

	public getPeriodStartTimestamp(timestamp: number, period: number = 1) {
		// const now = this.getUTCNowTimestamp();
		return Math.floor(timestamp / 60000 / period - 1) * 60000 * period;
	}

	public timestampToString(ts: number) {
		//just for debugging purpose
		return moment.utc(ts).format('YYYY-MM-DDTHH:mm:ss');
	}

	public getStatusProcess(tool: string, option: IOption) {
		let type = '';
		const platform = option.azure ? '_AZURE' : option.gcp ? '_GCP' : '_AWS';
		console.log('dynamo', option.dynamo);
		const privacy = option.dynamo ? '_PUBLIC' : '_PRIVATE';
		let source = '';

		switch (tool) {
			case CST.TRADES:
				type = 'TRADE';
				source = option.source ? '_' + option.source.toUpperCase() : '';
				break;
			case CST.TRIGGER:
			case CST.FETCH_EVENTS:
				type = 'EVENT';
				source =
					'_' +
					([CST.EVENT_START_PRE_RESET, CST.EVENT_START_RESET].includes(option.event)
						? option.event.toUpperCase()
						: 'OTHERS');
				break;
			case CST.COMMIT:
				type = 'FEED';
				break;
			case CST.CLEAN_DB:
				type = 'CLEANDB';
				break;
			case CST.DB_PRICES:
				type = 'PRICE';
				if (option.period === 1) source = '_MINUTELY';
				else if (option.period === 60) source = '_HOURLY';
				else source = '';
				break;
			case 'minutely':
				type = 'MINUTELY';
				break;
			case CST.FETCH_PRICE:
				type = 'FETCH';
				break;
			default:
				return '';
		}

		return type + platform + privacy + source;
	}

	public getUTCNowTimestamp() {
		return moment().valueOf();
	}

	public isNumber(input: any): boolean {
		const num = Number(input);
		return isFinite(num) && !isNaN(num);
	}

	public isEmptyObject(obj: object | undefined | null): boolean {
		if (!obj) return true;

		for (const prop in obj) if (obj.hasOwnProperty(prop)) return false;

		return true;
	}

	public async sleep(ms: number) {
		return new Promise(resolve => {
			setTimeout(resolve, ms);
		});
	}
}

const util = new Util();
export default util;
