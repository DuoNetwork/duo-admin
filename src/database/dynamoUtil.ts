import AWS from 'aws-sdk';
import { PutItemInput } from 'aws-sdk/clients/dynamodb';
import moment from 'moment';
import * as CST from '../constants';
import { IPrice, IPriceBar, ITrade } from '../types';
import util from '../util';

export class DynamoUtil {
	private ddb: undefined | AWS.DynamoDB = undefined;
	// private role: string = '';
	private process: string = 'UNKNOWN';
	private live: boolean = false;
	public init(live: boolean, role: string, process: string) {
		this.live = live;
		this.process = process;
		// this.role = role;
		AWS.config.loadFromPath('./src/keys/aws/' + (live ? 'live' : 'dev') + '/' + role + '.json');
		this.ddb = new AWS.DynamoDB({ apiVersion: CST.AWS_DYNAMO_API_VERSION });
	}

	public insertData(params: PutItemInput): Promise<void> {
		return new Promise(
			(resolve, reject) =>
				this.ddb
					? this.ddb.putItem(params, err => (err ? reject(err) : resolve()))
					: reject('dynamo db connection is not initialized')
		);
	}

	public convertTradeToSchema(trade: ITrade, systime: number) {
		return {
			[CST.DB_TX_ID]: { S: trade.id },
			[CST.DB_TX_PRICE]: { N: trade.price.toString() },
			[CST.DB_TX_AMOUNT]: { N: trade.amount.toString() },
			[CST.DB_TX_TS]: { N: trade.timestamp + '' },
			[CST.DB_TX_SYSTIME]: { N: systime + '' }
		};
	}

	public convertPriceToSchema(price: IPrice) {
		return {
			[CST.DB_HISTORY_PRICE]: { N: price.price + '' },
			[CST.DB_HISTORY_TIMESTAMP]: { N: price.timestamp + '' },
			[CST.DB_HISTORY_VOLUME]: { N: price.volume + '' }
		};
	}

	public convertPriceBarToSchema(priceBar: IPriceBar) {
		return {
			[CST.DB_HR_SRC_DATE]: { S: priceBar.source + '|' + priceBar.date },
			[CST.DB_HR_HOUR]: { N: priceBar.hour + '' },
			[CST.DB_HR_OPEN]: { N: priceBar.open + '' },
			[CST.DB_HR_HIGH]: { N: priceBar.high + '' },
			[CST.DB_HR_LOW]: { N: priceBar.low + '' },
			[CST.DB_HR_CLOSE]: { N: priceBar.close + '' },
			[CST.DB_HR_VOLUME]: { N: priceBar.volume + '' },
			[CST.DB_HR_TS]: { N: priceBar.timestamp + '' }
		};
	}

	public async insertTradeData(trade: ITrade, insertStatus: boolean): Promise<void> {
		const systemTimestamp = util.getNowTimestamp(); // record down the MTS
		const data = this.convertTradeToSchema(trade, systemTimestamp);

		const params = {
			TableName: this.live ? CST.DB_AWS_TRADES_LIVE : CST.DB_AWS_TRADES_DEV,
			Item: {
				[CST.DB_TX_SRC_DATE]: {
					S: trade.source + '|' + moment.utc(trade.timestamp).format('YYYY-MM-DD')
				},
				...data
			}
		};

		await this.insertData(params);
		if (insertStatus) await this.insertStatusData(data);
	}

	public async insertHourlyData(priceBar: IPriceBar): Promise<void> {
		return this.insertData({
			TableName: this.live ? CST.DB_AWS_HOURLY_LIVE : CST.DB_AWS_HOURLY_DEV,
			Item: {
				...this.convertPriceBarToSchema(priceBar)
			}
		});
	}

	public insertHeartbeat(): Promise<void> {
		return this.insertData({
			TableName: this.live ? CST.DB_AWS_STATUS_LIVE : CST.DB_AWS_STATUS_DEV,
			Item: {
				[CST.DB_ST_PROCESS]: {
					S: this.process
				},
				[CST.DB_ST_TS]: { N: util.getNowTimestamp() + '' }
			}
		});
	}

	public insertStatusData(data: object): Promise<void> {
		return this.insertData({
			TableName: this.live ? CST.DB_AWS_STATUS_LIVE : CST.DB_AWS_STATUS_DEV,
			Item: {
				[CST.DB_ST_PROCESS]: {
					S: this.process
				},
				...data
			}
		});
	}

	public readTradeData(source: string, date: string): Promise<any> {
		const params = {
			TableName: this.live ? CST.DB_AWS_TRADES_LIVE : CST.DB_AWS_TRADES_DEV,
			KeyConditionExpression: CST.DB_TX_SRC_DATE + ' = :' + CST.DB_TX_SRC_DATE,
			ExpressionAttributeValues: {
				[':' + CST.DB_TX_SRC_DATE]: { S: source + '|' + date }
			}
		};

		console.log(params);

		return new Promise(
			(resolve, reject) =>
				this.ddb
					? this.ddb.query(params, (err, data) => (err ? reject(err) : resolve(data)))
					: reject('dynamo db connection is not initialized')
		);
	}
}

const dynamoUtil = new DynamoUtil();
export default dynamoUtil;
