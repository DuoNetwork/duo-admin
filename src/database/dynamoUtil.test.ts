// import * as CST from '../constants';
import util from '../util';
import dynamoUtil from './dynamoUtil';
const sampleTrades = require('../samples/dynamoTrades.json');
const sampleMinutely = require('../samples/dynamoMinutely.json');

const trade = {
	source: 'src',
	id: 'id',
	price: 123,
	amount: 456,
	timestamp: 1234567890
};

const priceBar = {
	source: 'src',
	date: 'YYYY-MM-DD',
	hour: '00',
	minute: 0,
	open: 1,
	high: 3,
	low: 0,
	close: 2,
	volume: 123,
	timestamp: 1234567890
};

const event = {
	type: 'type',
	id: 'id',
	blockHash: 'blockHash',
	blockNumber: 123,
	transactionHash: 'txHash',
	logStatus: 'logStatus',
	parameters: {
		test: 'test'
	},
	timestamp: 1234567890,
}

test('connection initalization', () =>
	dynamoUtil.insertData({} as any).catch(error => expect(error).toMatchSnapshot()));

test('convertTradeToDynamo', () =>
	expect(dynamoUtil.convertTradeToDynamo(trade, 123)).toMatchSnapshot());

test('convertTradeToDynamo', () =>
	expect(
		dynamoUtil.convertPriceToDynamo({
			price: 123,
			timestamp: 1234567890,
			volume: 456
		})
	).toMatchSnapshot());

test('convertPriceBarToDynamo', () =>
	expect(dynamoUtil.convertPriceBarToDynamo(priceBar)).toMatchSnapshot());

test('convertEventToDynamo', () =>
	expect(dynamoUtil.convertEventToDynamo(event, 9876543210)).toMatchSnapshot());

test('insertTradeData', async () => {
	dynamoUtil.insertData = jest.fn(() => Promise.resolve());
	util.getNowTimestamp = jest.fn(() => 0);
	await dynamoUtil.insertTradeData(trade, true);
	expect((dynamoUtil.insertData as jest.Mock<Promise<void>>).mock.calls.length).toBe(2);
	expect((dynamoUtil.insertData as jest.Mock<Promise<void>>).mock.calls[0][0]).toMatchSnapshot();
	expect((dynamoUtil.insertData as jest.Mock<Promise<void>>).mock.calls[1][0]).toMatchSnapshot();
});

test('insertMinutelyData', async () => {
	dynamoUtil.insertData = jest.fn(() => Promise.resolve());
	await dynamoUtil.insertMinutelyData(priceBar);
	expect((dynamoUtil.insertData as jest.Mock<Promise<void>>).mock.calls.length).toBe(1);
	expect((dynamoUtil.insertData as jest.Mock<Promise<void>>).mock.calls[0][0]).toMatchSnapshot();
});

test('insertHourlyData', async () => {
	dynamoUtil.insertData = jest.fn(() => Promise.resolve());
	await dynamoUtil.insertHourlyData(priceBar);
	expect((dynamoUtil.insertData as jest.Mock<Promise<void>>).mock.calls.length).toBe(1);
	expect((dynamoUtil.insertData as jest.Mock<Promise<void>>).mock.calls[0][0]).toMatchSnapshot();
});

test('insertEventData', async () => {
	dynamoUtil.insertData = jest.fn(() => Promise.resolve());
	await dynamoUtil.insertEventData([event]);
	expect((dynamoUtil.insertData as jest.Mock<Promise<void>>).mock.calls.length).toBe(1);
	expect((dynamoUtil.insertData as jest.Mock<Promise<void>>).mock.calls[0][0]).toMatchSnapshot();
});

test('insertHeartbeat', async () => {
	dynamoUtil.insertData = jest.fn(() => Promise.resolve());
	await dynamoUtil.insertHeartbeat();
	expect((dynamoUtil.insertData as jest.Mock<Promise<void>>).mock.calls.length).toBe(1);
	expect((dynamoUtil.insertData as jest.Mock<Promise<void>>).mock.calls[0][0]).toMatchSnapshot();
});

test('insertStatusData', async () => {
	dynamoUtil.insertData = jest.fn(() => Promise.resolve());
	await dynamoUtil.insertStatusData({ test: 'test' });
	expect((dynamoUtil.insertData as jest.Mock<Promise<void>>).mock.calls.length).toBe(1);
	expect((dynamoUtil.insertData as jest.Mock<Promise<void>>).mock.calls[0][0]).toMatchSnapshot();
});

test('readTradeData', async () => {
	dynamoUtil.queryData = jest.fn(() => Promise.resolve(sampleTrades));
	expect(await dynamoUtil.readTradeData('source', 'datetime')).toMatchSnapshot();
	expect((dynamoUtil.queryData as jest.Mock<Promise<void>>).mock.calls.length).toBe(1);
	expect((dynamoUtil.queryData as jest.Mock<Promise<void>>).mock.calls[0][0]).toMatchSnapshot();
});

test('readMinutelyData', async () => {
	dynamoUtil.queryData = jest.fn(() => Promise.resolve(sampleMinutely));
	expect(await dynamoUtil.readMinutelyData('source', 'datetime')).toMatchSnapshot();
	expect((dynamoUtil.queryData as jest.Mock<Promise<void>>).mock.calls.length).toBe(1);
	expect((dynamoUtil.queryData as jest.Mock<Promise<void>>).mock.calls[0][0]).toMatchSnapshot();
});
