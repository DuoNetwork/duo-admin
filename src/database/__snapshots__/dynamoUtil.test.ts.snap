// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`connection initalization 1`] = `"dynamo db connection is not initialized"`;

exports[`convertPriceBarToSchema 1`] = `
Object {
  "close": Object {
    "N": "2",
  },
  "high": Object {
    "N": "3",
  },
  "hour": Object {
    "N": "1",
  },
  "low": Object {
    "N": "0",
  },
  "open": Object {
    "N": "1",
  },
  "sourceDate": Object {
    "S": "src|YYYY-MM-DD",
  },
  "timestamp": Object {
    "N": "1234567890",
  },
  "volume": Object {
    "N": "123",
  },
}
`;

exports[`convertTradeToSchema 1`] = `
Object {
  "amount": Object {
    "N": "456",
  },
  "id": Object {
    "S": "id",
  },
  "price": Object {
    "N": "123",
  },
  "systime": Object {
    "N": "123",
  },
  "timestamp": Object {
    "N": "1234567890",
  },
}
`;

exports[`convertTradeToSchema 2`] = `
Object {
  "price": Object {
    "N": "123",
  },
  "timestamp": Object {
    "N": "1234567890",
  },
  "volume": Object {
    "N": "456",
  },
}
`;

exports[`insertHeartbeat 1`] = `
Object {
  "Item": Object {
    "process": Object {
      "S": "UNKNOWN",
    },
    "timestamp": Object {
      "N": "0",
    },
  },
  "TableName": "status_dev",
}
`;

exports[`insertHourlyData 1`] = `
Object {
  "Item": Object {
    "close": Object {
      "N": "2",
    },
    "high": Object {
      "N": "3",
    },
    "hour": Object {
      "N": "1",
    },
    "low": Object {
      "N": "0",
    },
    "open": Object {
      "N": "1",
    },
    "sourceDate": Object {
      "S": "src|YYYY-MM-DD",
    },
    "timestamp": Object {
      "N": "1234567890",
    },
    "volume": Object {
      "N": "123",
    },
  },
  "TableName": "hourly_dev",
}
`;

exports[`insertStatusData 1`] = `
Object {
  "Item": Object {
    "process": Object {
      "S": "UNKNOWN",
    },
    "test": "test",
  },
  "TableName": "status_dev",
}
`;

exports[`insertTradeData 1`] = `
Object {
  "Item": Object {
    "amount": Object {
      "N": "456",
    },
    "id": Object {
      "S": "id",
    },
    "price": Object {
      "N": "123",
    },
    "sourceDate": Object {
      "S": "src|1970-01-15",
    },
    "systime": Object {
      "N": "0",
    },
    "timestamp": Object {
      "N": "1234567890",
    },
  },
  "TableName": "trades_dev",
}
`;

exports[`insertTradeData 2`] = `
Object {
  "Item": Object {
    "amount": Object {
      "N": "456",
    },
    "id": Object {
      "S": "id",
    },
    "price": Object {
      "N": "123",
    },
    "process": Object {
      "S": "UNKNOWN",
    },
    "systime": Object {
      "N": "0",
    },
    "timestamp": Object {
      "N": "1234567890",
    },
  },
  "TableName": "status_dev",
}
`;
